src/main/java/ch/njol/skript/expressions/ExprEnchantmentOffer.java

```diff
@@ -0,0 +1,151 @@
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.Bukkit;
+import org.bukkit.enchantments.EnchantmentOffer;
+import org.bukkit.event.Event;
+import org.bukkit.event.enchantment.PrepareItemEnchantEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.log.ErrorQuality;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Enchantment Offer")
+@Description("The enchantment offer in enchant prepare events.")
+@Examples({"on enchant prepare:",
+		"\tsend \"Your enchantment offers are: %the enchantment offers%\" to player"})
+@Since("INSERT VERSION")
+@Events("enchant prepare")
+@RequiredPlugins("1.11 or newer")
+public class ExprEnchantmentOffer extends SimpleExpression<EnchantmentOffer> {
+
+	static {
+		if (Skript.classExists("org.bukkit.enchantments.EnchantmentOffer")) {
+			Skript.registerExpression(ExprEnchantmentOffer.class, EnchantmentOffer.class, ExpressionType.SIMPLE, 
+					"[the] enchant[ment] offers",
+					"enchant[ment] offer %number%",
+					"[the] %number%(st|nd|rd|th) enchant[ment] offer");
+		}
+	}
+
+	@SuppressWarnings("null")
+	private Expression<Number> exprOfferNumber;
+
+	private boolean multiple;
+
+	@SuppressWarnings({"null", "unchecked"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!ScriptLoader.isCurrentEvent(PrepareItemEnchantEvent.class)) {
+			Skript.error("Enchantment offers are only usable in enchant prepare events", ErrorQuality.SEMANTIC_ERROR);
+			return false;
+		}
+		if (matchedPattern == 0) {
+			multiple = true;
+		} else {
+			exprOfferNumber = (Expression<Number>) exprs[0];
+			multiple = false;
+		}
+		return true;
+	}
+
+	@SuppressWarnings("null")
+	@Override
+	@Nullable
+	protected EnchantmentOffer[] get(Event e) {
+		if (multiple)
+			return ((PrepareItemEnchantEvent) e).getOffers();
+
+		int offerNumber = exprOfferNumber.getSingle(e).intValue();
+		if (offerNumber < 1 || offerNumber > 3)
+			return new EnchantmentOffer[]{};
```

Comment:
new EnchantmentOffer[0];

Reply:
also wouldn't it make more sense to just check the length of the offer array? just because this is the length now doesn't mean it will be in the future

Reply:
I will have to change that. There could be 0-3 offers, so I'll have to account for that.